scripts/package/fpsci_packager_generator.py

```diff
@@ -0,0 +1,125 @@
+
+if __name__ == '__main__':
+    # The log file which lists the files used
+    log = open(inputLog)
+
+    # Get the current set of lines in the script
+    lineSet = set(open(outputScript))
+    for line in sorted(lineSet):
+        if line.startswith('#'):
+            lineSet.remove(line)
+
+    # Hard coded files to copy
+    lineSet.add('mkdir -p ' + distPath + '\n')
+    lineSet.add('cp data-files/g3d-license.txt '  + distPath + '\n')
+    lineSet.add('cp LICENSE.txt '  + distPath + '\n')
+    lineSet.add('cp README.txt '  + distPath + '\n')
+    # For some reason these files aren't captured by G3D as used
+    lineSet.add('mkdir -p dist/shader/DefaultRenderer/\n')
+    lineSet.add('cp c:/g3d/G3D10/data-files/shader/DefaultRenderer/DefaultRenderer_OIT_writePixel.glsl dist/shader/DefaultRenderer/\n')
+    lineSet.add('mkdir -p dist/shader/UniversalSurface/\n')
+    lineSet.add('cp c:/g3d/G3D10/data-files/shader/UniversalSurface/UniversalSurface_depthPeel.pix dist/shader/UniversalSurface/\n')
+
+    # Set up copy of .exe
+    lineSet.add('cp Build/FirstPersonScience-x64-Release/FirstPersonScience.exe ' + distPath + '\n')
+
+    # main loop, need to ignore lines before the file list begins
+    beforeFiles = True
+    for line in log.readlines():
+        # In the file list, add this file to what we want to copy
+        if not beforeFiles and line != '\n':
+            filename = line.strip()
+            basename = filename.split('/')[-1]
+
+            # Exclude the lists of files we don't want to package
+            if basename in excludeList or basename in secondaryExcludeList:
+                continue
+            
+            # Absolute path given
+            if filename[1:].startswith(':/'):
+                for path in basePath:
+                    dest = filename.find(path)
+                    if dest > 0:
+                        dest += len(path) + 1
+                        additionalPath = filename[dest:-len(basename)]
+                        # if this file is inside a compressed structure, skip copying it
+                        if additionalPath.find('.pk3') > 0 or additionalPath.find('.zip') > 0:
+                            break
+
+                        # Make paths generic
+                        filename = filename.replace(g3dPath, "$g3d").replace(runPath, ".")
+
+                        # Write commands to file
+                        lineSet.add('mkdir -p ' + distPath + additionalPath + '\n')
+                        lineSet.add('cp ' + filename + ' ' + distPath + additionalPath + '\n')
+
+            # file is in the PATH, need to find it
+            else:
+                lineSet.add('cp `which ' + filename + '` ' + distPath + '\n')
+
+        # Find where the filenames start
+        if line.startswith('    ###    Files Used    ###'):
+            beforeFiles = False
+    log.close()
+
+    # Write the set of commands in sorted order
+    with open(outputScript, 'w') as packageScript:
```

Comment:
I agree that we should figure out the right way to support experiment builds. My current thinking is that the experiment would exist as a branch/fork of FPSci and that it would have scripts that diverge for that one experiment. We should open an issue to potentially solve that issue more generally in the future. In this pull request, I'm focused on the general FPSci build process.